internal/config/cgmgr/cgroupfs_linux.go

```diff
@@ -90,25 +90,25 @@ func (m *CgroupfsManager) ContainerCgroupStats(sbParent, containerID string) (cg
 	}
 	cgstats = &CgroupStats{
 		MostStats:     libCtrStats,
-		OtherMemStats: generateOtherMemoryStats(libCtrStats.MemoryStats),
+		OtherMemStats: generateOtherMemoryStats(&libCtrStats.MemoryStats),
 		SystemNano:    time.Now().UnixNano(),
 	}
 	return cgstats, nil
 }
 
 // GetCtrCgroupManager takes the cgroup parent, and container ID.
 // It returns the raw libcontainer cgroup manager for that container.
-func (m *CgroupfsManager) GetCtrCgroupManager(sbParent, containerID string) (cgMgr libctrCg.Manager, err error) {
+func (m *CgroupfsManager) GetCtrCgroupManager(sbParent, containerID string) (libctrCg.Manager, error) {
 	if !node.CgroupIsV2() {
-		if cgMgr = m.v1CtrCgMgr[containerID]; cgMgr != nil {
+		if cgMgr := m.v1CtrCgMgr[containerID]; cgMgr != nil {
```

Comment:
so this will create a new entry in the map, you need to do `cgMgr, ok := ..; ok && cgMgr != nil {` to not add an entry

Reply:
oh shoot sorry I commented this on the wrong PR, lemme move this convo over there